### DISCUSSION TITLE
Does data/icon_cache store favicons for all users?

### DISCUSSION THREAD

**Discussion by Jinra** [dani-garcia/vaultwarden#3551] [Q&A] [answered]
I'm curious how this folder works for caching icons. If it behaves as I suspect, which is storing icons for any user with a vault, then it seems like a security issue that container admins are able to view this in the container file system. I only just spun up my own instance and noticed this is populated with icons from my vault after import, so I can't confirm any other users having their icons in the folder yet.

**Comment by BlackDex** [comment, marked as answer]
Icon downloading is a _shared_ feature, so it downloads for all users.
The only way to fully mitigate your concern is to disable icon downloading on the client side for all clients you have.

Else they still create a call to the server, which is logged by probably your reverse proxy and Vaultwarden. Doesn't matter if you use the redirect feature or disable it on the server side.

**Reply by Jinra** [reply]
Ah, that's unfortunate, thanks for the answer. I can see this being a minor concern for my users as I would be able to see what sites they have entries for to a limited extent.
